### PR TITLE
PLAT-48662: switch to use node net instead of native nc by default

### PR DESCRIPTION
**Pull request by c3wenjiaowang**
more description of the issue we observed can be found here: https://c3energy.atlassian.net/browse/PLAT-48662

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ function nodeNetCatSrc(port, input) {
   );
 }
 
-const FUNCTION_PRIORITY = [nativeNC, nodeNC];
+// Use nodeNC to avoid native nc error (spawnSync error observed with msc os 11.2.3)
```
Comment (c3-tiffanyhui) [inline lib/index.js]: @c3wenjiaowang  is there a spawnSync issue that you can link here as well (one that describes OS bug) or its not online and you just happened to try it /  notice it

**Review by tonescones — APPROVED** [review]

File: lib/index.js (hunk 1 above)
Comment (c3wenjiaowang) [reply lib/index.js]: @c3-tiffanyhui Didn't find any issue reported online, just figured with the 3 colleagues who reported this issue, and they are all on mac os 11.2.3...

**Comment by tonescones** [comment]
@c3wenjiaowang don't you need to increase the version of this package too? just in case this breaks something else.

**Comment by c3wenjiaowang** [comment]
@tonescones Thanks for the reminder! Yes, we should. I have just reverted the merge in `main` branch, so that Alex can upgrade his mac os version tonight and verify whether it resolves the issue. When I merge again afterwards, I will upgrade the package version.

**Pull request merged**